tests/foreman/api/test_computeresource_gce.py

```diff
@@ -0,0 +1,250 @@
+            project=GCE_SETTINGS['project_id'],
+            zone=GCE_SETTINGS['zone'],
+            organization=[self.org],
+            location=[self.loc]
+        ).create()
+        assert compresource.name == cr_name
+        # Testing Update
+        new_name = gen_string('alpha')
+        description = gen_string('utf8')
+        new_zone = random.choice(VALID_GCE_ZONES)
+        compresource.name = new_name
+        compresource.description = description
+        compresource.zone = new_zone
+        compresource.update(['name', 'description', 'zone'])
+        # Testing Read
+        updated = entities.GCEComputeResource(id=compresource.id).read()
+        assert updated.name == new_name
+        assert updated.description == description
+        assert updated.zone == new_zone
+        # Testing Delete
+        updated.delete()
+        assert not entities.GCEComputeResource().search(
+            query={'search': 'name={}'.format(new_name)})
+
+    @tier3
+    def test_positive_check_available_images(self):
+        """Verify all the images from GCE are available to select from
+
+        :id:
+
+        :expectedresults: All the images from Google CR should be available to select in GCE CR
+
+        :CaseLevel: Integration
+        """
+        assert len(
+            GCEnvironment.gce_compute().read().available_images()
+        ) == len(gceclient.list_templates(True))
+
+    @tier3
+    def test_positive_check_available_networks(self):
+        """Verify all the networks from GCE are available to select
+
+        :id:
+
+        :expectedresults: All the networks from Google CR should be available to select in GCE CR
+
+        :CaseLevel: Integration
+        """
+        assert len(
+            GCEnvironment.gce_compute().available_networks()
+        ) == len(
+            self.gclient.list_network()
+        )
+
+    @tier3
+    def test_positive_check_available_flavors(self):
+        """Verify flavors from GCE are available to select
+
+        :id:
+
+        :expectedresults: All the flavors from Google CR should be available to select in GCE Host
+
+        :CaseLevel: Integration
+        """
+        assert len(GCEnvironment.gce_compute().available_flavors()) > 1
+
+    @tier3
+    def test_positive_create_finish_template_image(self):
+        """Finish template image along with username is being added in GCE CR
+
+        :id:
+
+        :steps:
+            1. Create a GCE Compute Resource.
+            2. Add a finish template based image in GCE CR.
+
+        :expectedresults: Finish template image should be added in GCE CR along with username
+
+        :CaseLevel: Integration
+        """
+        assert GCEnvironment.finishimg(
+            gceclient).compute_resource.id == GCEnvironment.gce_compute().id
+        assert GCEnvironment.finishimg(gceclient).uuid == GCEnvironment.latest_rhel_uuid()
+        assert GCEnvironment.finishimg().username == GCEnvironment.finishimg_user
```

Comment:
This is one of weaknesses of that wrapper and API - it's not clear where test data comes from and what exactly is verified here. Since `GCEnvironment` controls both `finishimg` and `finishimg_user`, it looks more like unit test for wrapper that verifies if certain data is available through two different interfaces.

Reply:
That can be fixed by passing a username parameter. But That way I will end up with multiple usernames and this being a helper function which sets only once, I dont want to do that.